src/sqlfluff/core/rules/functional/templated_file_slices.py

```diff
@@ -35,3 +35,25 @@ def any(
             if predicate is None or predicate(s):
                 return True
         return False
+
+    def select(
+        self,
+        select_if: Optional[Callable[[TemplatedFileSlice], bool]] = None,
+        loop_while: Optional[Callable[[TemplatedFileSlice], bool]] = None,
+        start_slice: Optional[TemplatedFileSlice] = None,
+        stop_slice: Optional[TemplatedFileSlice] = None,
+    ) -> "TemplatedFileSlices":  # pragma: no cover
+        """Retrieve range/subset.
+
+        NOTE: Iterates the slices BETWEEN start_slice and stop_slice, i.e. those
+        slices are not included in the loop.
+        """
+        start_index = self.index(start_slice) if start_slice else -1
+        stop_index = self.index(stop_slice) if stop_slice else len(self)
+        buff = []
+        for slice_ in self[start_index + 1 : stop_index]:
+            if loop_while is not None and not loop_while(slice_):
+                break
+            if select_if is None or select_if(slice_):
+                buff.append(slice_)
+        return TemplatedFileSlices(*buff, templated_file=self.templated_file)
```

Comment:
Added this method, equivalent to the `select()` methods on `Segments` and `RawFileSlices`. Ended up not using it, but thought it was worth keeping for future use.